source/fuzz/fuzzer_pass_permute_function_variables.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SOURCE_FUZZ_FUZZER_PASS_PERMUTE_FUNCTION_VARIABLES_H_
+#define SOURCE_FUZZ_FUZZER_PASS_PERMUTE_FUNCTION_VARIABLES_H_
+
+#include "source/fuzz/fuzzer_pass.h"
+#include "source/fuzz/transformation_swap_function_variables.h"
+
+namespace spvtools {
+namespace fuzz {
+
+// A pass that permute function variables.
+// this is final class so, can't be inherited. 
```

Comment:
Start a new sentence with a capital letter: "T".

However, try to avoid comments that simply repeat what the code says. "final" means "cannot have subclasses", so a reader of the code will know that this class cannot be subclassed; there's no need to point that out with an extra comment. (Note: sometimes, for really unusual C++ features, I do advise spelling out what they mean with a comment, but not for simple features like "final").